### DISCUSSION TITLE
OBS Browser Source activation detection

### DISCUSSION THREAD

**Discussion by IGunzl** [WarmUpTill/SceneSwitcher#706] [Q&A] [answered]
Good evening,
I have a Browser Source that, from time to time, shows content on the screen, similar to a donation message, but without any sound.
I am trying to set up a Macro that looks at my source and when something appears, it will change the scene.
Basically, my source is always active, but not showing anything and once it will show something I need my scene to switch.
Source Macros are offering “when is active”, “when is showing” and “settings”, but my source is always active and showing, so macros launch themselves in a loop. Also my source settings are always the same.
In case it could help, there is a possibility to use data provided by the source with a websocket, and I have a link to it, but I have no idea how it works.
Is there a way to make it work or will it ever be added to the plug-in?
Thank you in advance and have a nice day.

**Comment by WarmUpTill** [comment, marked as answer]
Maybe using the "[Video](https://github.com/WarmUpTill/SceneSwitcher/wiki/Video-condition)" condition is an option here?

**Reply by IGunzl** [reply]
Thank you very much, it is working as a video source. I was not even thinking that Video stands also for this, because I was considering it as a Media condition.
Also, right now I noticed a certain bug that crashes OBS every single time an action is performed. I have the last version of OBS - 29.0.2 (64 bit, Windows 10 Home with the last updates installed) and ASW - 1.20.1.
You choose “If - Video - Source - Name - has changed” and mark “Perform check only in area” and then click on “Select area”. If you try to move the lateral scrollbar down and then begin to select the area, OBS will crash. You are still able to select without touching anything, but sometimes it crashes even simply highlighting the area.
Still, it worked, so thank you for your support and have a nice weekend.

**Reply by WarmUpTill** [reply]
Glad to hear that it is working! :)
The particular crash you mentioned was fixed a few releases ago if I am not mistaken.

**Reply by IGunzl** [reply]
Oh, I see, I will update it then
